File: backend/routers/Main.router.go
package routers

import (
	"github.com/gin-gonic/gin"
	"github.com/go-redis/redis/v8"
	"github.com/jmoiron/sqlx"
)

func BootstrapRouter(db *sqlx.DB, redisConn *redis.Client) *gin.Engine {
	app := gin.Default()

	// Префиксный роутер
	prefixRouter := app.Group("/backend")
	prefixRouter.GET("/", func(ctx *gin.Context) {
		ctx.JSON(200, map[string]any{
			"message": "Blog Index message...",
		})
	})

	// Пользователи
	userRouter := prefixRouter.Group("/users")
	setupUserRouter(userRouter, db, redisConn)

	// Посты
	postsRouter := prefixRouter.Group("/posts")
	setupPostRouter(postsRouter, db, redisConn)

	return app
}

File: backend/src/routes/Main.router.ts
import { Router } from "express";
import { PostRouter } from "./Post.router";
import { UserRouter } from "./User.router";


export const MainRouter = Router()
const prefixRouter = Router()

MainRouter.use('/backend', prefixRouter)

prefixRouter.use('/users', UserRouter)
prefixRouter.use('/posts', PostRouter)
